cmd/compute-domain-kubelet-plugin/computedomain.go

```diff
@@ -339,14 +341,47 @@ func (m *ComputeDomainManager) GetComputeDomain(ctx context.Context, cdUID strin
 	return cd, nil
 }
 
+// removeStaleNodeLabel() reads the current CD label value for this node. If a
+// corresponding ComputeDomain is not found, the label is assumed to be stale
+// and is removed. Any errors are logged, but not returned.
+func (m *ComputeDomainManager) removeStaleNodeLabel(ctx context.Context) {
+	node, err := m.config.clientsets.Core.CoreV1().Nodes().Get(ctx, m.config.flags.nodeName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("error getting node data: %v", err)
+		return
+	}
+
+	if _, exists := node.Labels[computeDomainLabelKey]; exists {
+		return
+	}
+
+	uid := node.Labels[computeDomainLabelKey]
+	cd, err := m.GetComputeDomain(ctx, uid)
+	if err != nil {
+		klog.Errorf("error getting ComputeDomain: %v", err)
+		return
+	}
+
+	if cd != nil {
+		// CD still exists, do not clean up
+		return
+	}
+
+	if err := m.RemoveNodeLabel(ctx, uid); err != nil {
+		klog.Errorf("error removing node label for ComputeDomain '%s': %v", uid, err)
+	}
+}
+
 func (m *ComputeDomainManager) periodicCleanup(ctx context.Context) {
 	ticker := time.NewTicker(cleanupInterval)
 	defer ticker.Stop()
 
 	for {
 		select {
 		case <-ticker.C:
-			klog.V(6).Infof("Running periodic sync to remove artifacts owned by stale ComputeDomain")
+			klog.V(6).Infof("Running periodic cleanup to remove stale ComputeDomain artifacts")
+
+			m.removeStaleNodeLabel(ctx)
```

Comment:
Can maybe give this a better name: something like `removeStaleNodeLabelIfExists()`?